freqtrade/data/dataprovider.py

```diff
@@ -63,9 +80,113 @@ def _set_cached_df(
         :param dataframe: analyzed dataframe
         :param candle_type: Any of the enum CandleType (must match trading mode!)
         """
-        self.__cached_pairs[(pair, timeframe, candle_type)] = (
+        pair_key = (pair, timeframe, candle_type)
+        self.__cached_pairs[pair_key] = (
             dataframe, datetime.now(timezone.utc))
 
+    # For multiple producers we will want to merge the pairlists instead of overwriting
+    def _set_producer_pairs(self, pairlist: List[str], producer_name: str = "default"):
+        """
+        Set the pairs received to later be used.
+        This only supports 1 Producer right now.
+
+        :param pairlist: List of pairs
+        """
+        self.__producer_pairs[producer_name] = pairlist.copy()
```

Comment:
I think the `.copy()` needs to be moved to the getter for this.

we don't do anything with the pairlist after setting it (so no risk of modification) - but a consumer may drop pairs out of it - modifying the "datastore".